pkg/postgres/query/builder.go

```diff
@@ -50,20 +50,19 @@ func (qb *Builder) AddFilterRequest(request *filter.Request) error {
 		var (
 			err               error
 			valueIsList       bool
-			valueList         []any
 			conditionTemplate string
 			conditionParams   []any
 		)
-		valueIsList, valueList, err = checkFieldValueType(field)
+		valueIsList, _, err = checkFieldValueType(field)
 		if err != nil {
 			return fmt.Errorf("error checking filter field value type '%s': %w", field, err)
 		}
 		conditionTemplate, conditionParams, err =
-			composeQuery(qb.querySettings.FilterFieldMapping, field, valueIsList, valueList)
+			composeQuery(qb.querySettings.FilterFieldMapping, field, valueIsList)
 		if err != nil {
 			return fmt.Errorf("error composing query from filter field %w", err)
 		}
-		if index == len(request.Fields)-1 {
+		if index == len(request.Fields)-1 && len(request.Fields) != 1 {
```

Comment:
```suggestion
		if index > 0 {
```
I think the logic operator needs to be prepended to every element, except the first one

Reply:
Thanks, suggested change has been added